src/main/java/io/stargate/sgv2/jsonapi/exception/JsonApiException.java

```diff
@@ -106,4 +135,134 @@ public ErrorCode getErrorCode() {
   public Response.Status getHttpStatus() {
     return httpStatus;
   }
+
+  private String getErrorFamily() {
+    // some error codes should be classified as SERVER errors but do not have any pattern
+    Set<ErrorCode> serverFamily =
+        new HashSet<>() {
+          {
+            add(COUNT_READ_FAILED);
+            add(CONCURRENCY_FAILURE);
+            add(TOO_MANY_COLLECTIONS);
+            add(VECTOR_SEARCH_NOT_AVAILABLE);
+            add(VECTOR_SEARCH_NOT_SUPPORTED);
+            add(VECTORIZE_FEATURE_NOT_AVAILABLE);
+            add(VECTORIZE_SERVICE_NOT_REGISTERED);
+            add(VECTORIZE_SERVICE_TYPE_UNAVAILABLE);
+            add(VECTORIZE_INVALID_AUTHENTICATION_TYPE);
+            add(VECTORIZE_CREDENTIAL_INVALID);
+            add(VECTORIZECONFIG_CHECK_FAIL);
+            add(UNAUTHENTICATED_REQUEST);
+            add(COLLECTION_CREATION_ERROR);
+            add(INVALID_QUERY);
+            add(NO_INDEX_ERROR);
+          }
+        };
+
+    if (serverFamily.contains(errorCode)
+        || errorCode.name().startsWith("SERVER")
+        || errorCode.name().startsWith("EMBEDDING")) {
+      return "SERVER";
+    }
+    return "REQUEST";
+  }
+
+  private String getErrorScope(String family) {
```

Comment:
Same as with family, seems better to pass explicit info either by caller (eventually), or from `ErrorCode` (at least initially).

Reply:
as above, we want to remove all the old error code in 2 to 3 weeks so this to reduce the number of changed lines. 

Hazel - this could be a Map<Set<ErrorCode> , String> (string for the scope) created as static final so you . If you do that you iterate  the entries and test the keys, it is a little less code than a Map<ErrorCode, String> and better conveys what we are doing.